### ISSUE TITLE
Removing return type annotation for async functions

### ISSUE THREAD

**Issue by ivov** [n8n-io/n8n#671]
**Describe the bug**
The codebase contains ~150 instances of the following pattern, where the return type is annotated as `Promise<any>` and immediately after the TSLint exception `// tslint:disable-line:no-any` is included to skip over the `any` contained in the `Promise` object.

```
export async function acuitySchedulingApiRequest(this: IHookFunctions | IExecuteFunctions | IExecuteSingleFunctions | ILoadOptionsFunctions | IWebhookFunctions, method: string, resource: string, body: any = {}, qs: IDataObject = {}, uri?: string, option: IDataObject = {}): Promise<any> { // tslint:disable-line:no-any
```

**To Reproduce**
Search for this regex pattern in the codebase: `: Promise<any> \{ // tslint:disable-line:no-any`

**Expected behavior**
Every async function returns a `Promise`, [always](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function). Since the functions in which this pattern occurs are all async, the return type annotation `: Promise<any>` is not necessary. TypeScript infers that a `Promise` will be returned. Removing the return type annotation allows us to also remove the TSLint exception, making the code much more readable, without losing type information.

With `Promise<any>` (current)
![1](https://user-images.githubusercontent.com/44588767/84702977-db8c1000-af2d-11ea-81f4-1941dcfc83a9.png)

Without `Promise<any>` and without TSlint exception (proposed)
![2](https://user-images.githubusercontent.com/44588767/84703018-e5ae0e80-af2d-11ea-9b6a-4562c90c7c91.png)

**Environment (please complete the following information):**
- OS: Windows 8.1
- n8n Version: 0.69.0
- Node.js Version: 12.18.0

**Additional context**
@janober clarified that this pattern can be fixed "in all places which implements a function which is defined somewhere else it is fine. But not in the main definition that people can actually see that it is the expected response type." Most instances of this pattern are in `GenericFunctions.ts`. I will be especially checking core files to ensure that no type information is lost, but as long as it is an async function, the same logic for removing `Promise<any>` should apply.

**Comment by ivov** [comment]
I kept digging and found the following.

Currently, with `Promise<any>`, the functions in `GenericFunctions.ts` files return a `Promise` that, when awaited, becomes simply `any`. This `any` result is then used in the object that is pushed to `returnData`. In the below example, `responseData` is `any` and so are its related values `responseData.items` and `baseData`.

![withPromise1](https://user-images.githubusercontent.com/44588767/84844232-503f7700-b020-11ea-8dc0-e1a8c1f9f3d6.png)

![withPromise2](https://user-images.githubusercontent.com/44588767/84844965-f8097480-b021-11ea-80f1-1353695dff47.png)

Now, removing the `Promise<any>` annotation means that the function will return a `Promise` that, when awaited, becomes the correct type, in this case `{ items: IDataObject[] }`.

![withoutPromise1](https://user-images.githubusercontent.com/44588767/84844272-5fbec000-b020-11ea-9f59-c51bba127644.png)

![withoutPromise2](https://user-images.githubusercontent.com/44588767/84844859-b8428d00-b021-11ea-9fd6-8d3e4e8b390e.png)

This is good, as the function should return the proper type.

But this also reveals a mismatch. The object of type `IDataObject` is looped over to fill in the values of the object pushed into `returnData`. The values `name` and `value` of that object expect only `string` and `string | number` respectively, whereas the values in `IDataObject` include a wide range of other types, as they can be `GenericValue`, which stands for six primitive types.

```
export interface IDataObject {
    [key: string]: GenericValue | IDataObject | GenericValue[] | IDataObject[];
}

export declare type GenericValue = string | object | number | boolean | undefined | null;
```

In short, removing `Promise<any>` and its TSLint exception provides much better type information and allows for less code, but this requires a core type refactoring. (Alternatively, casting the values is possible, but it would involve a lot of effort for only a temporary fix.) 

`any` is best avoided, especially in often used functions, so this type mismatch should be addressed. But given the extent of the changes (133 instances in 76 files), I think that the benefits of making this change probably do not justify a refactoring for the time being.

**Comment by janober** [comment]
Thanks a lot for putting this together @ivov! I totally agree with you that it should be changed but is currently not the highest priority.

**Comment by krynble** [comment]
Closing this in favor of another discussion on Jira.
